rfcs/text/0015_ols_phase_1.md

```diff
@@ -0,0 +1,268 @@
+
+## 3.1 Saved Objects Service
+
+### 3.1.1 Type registry
+The [saved objects type registry](https://github.com/elastic/kibana/blob/701697cc4a34d07c0508c3bdf01dca6f9d40a636/src/core/server/saved_objects/saved_objects_type_registry.ts) will allow consumers to register "private" saved object types via a new `accessClassification` property:
+
+```ts
+/**
+ * The accessClassification dictates the protection level of the saved object:
+ *  * public (default): instances of this saved object type will be accessible to all users within the given namespace, who are authorized to act on objects of this type.
+ *  * private: instances of this saved object type will belong to the user who created them, and will not be accessible by other users, except for administrators.
+ */
+export type SavedObjectsAccessClassification = 'public' | 'private';
+
+// Note: some existing properties have been omitted for brevity.
+export interface SavedObjectsType {
+  name: string;
+  hidden: boolean;
+  namespaceType: SavedObjectsNamespaceType;
+  mappings: SavedObjectsTypeMappingDefinition;
+
+  /**
+   * The {@link SavedObjectsAccessClassification | accessClassification} for the type.
+   */
+  accessClassification?: SavedObjectsAccessClassification;
+}
+
+// Example consumer
+class MyPlugin {
+  setup(core: CoreSetup) {
+    core.savedObjects.registerType({
+      name: 'user-settings',
+      accessClassification: 'private',
+      namespaceType: 'single',
+      hidden: false,
+      mappings,
+    })
+  }
+}
+```
+
+### 3.1.2 Schema
+Saved object ownership will be recorded as metadata within each `private` saved object. We do so by adding a top-level `accessControl` object with a singular `owner` property:
+
+```ts
+/**
+ * Describes which users should be authorized to access this SavedObject.
+ *
+ * @public
+ */
+export interface SavedObjectAccessControl {
+  /** The owner of this SavedObject. */
+  owner: string;
+}
+
+// Note: some existing fields have been omitted for brevity
+export interface SavedObject<T = unknown> {
+  id: string;
+  type: string;
+  attributes: T;
+  references: SavedObjectReference[];
+  namespaces?: string[];
+  /** Describes which users should be authorized to access this SavedObject. */
+  accessControl?: SavedObjectAccessControl;
+}
+```
+
+### 3.1.3 Saved Objects Client: Security wrapper
+
+The [security wrapper](https://github.com/elastic/kibana/blob/701697cc4a34d07c0508c3bdf01dca6f9d40a636/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts) authorizes and audits operations against saved objects.
+
+There are two primary changes to this wrapper:
+
+#### Attaching Access Controls
+
+This wrapper will be responsible for attaching an access control specification to all private objects before they are created.
+It will also allow users to provide their own access control specification in order to support the import/create use cases.
```

Comment:
__question:__ how exactly is this supposed to support import use case? You mean it will provide a way to specify what owner should be set for all `private` objects during import?

Reply:
Each private object will have its access specification attached during export, so that it may be preserved on import

Reply:
So, in theory, users can modify this specification before importing SO? It might be acceptable, though.